lib/db_utils.go

```diff
@@ -10661,6 +10681,409 @@ func DBDeletePostAssociationWithTxn(txn *badger.Txn, snap *Snapshot, association
+	return data
+}
+
+// LockedBalanceEntry Put/Delete Operations (Badger Writes)
+
+func DbPutLockedBalanceEntryMappingsWithTxn(txn *badger.Txn, snap *Snapshot, blockHeight uint64,
+	lockedBalanceEntry LockedBalanceEntry) error {
+	// Sanity check the fields in the LockedBalanceEntry used in constructing the key.
+	if len(lockedBalanceEntry.HODLerPKID) != btcec.PubKeyBytesLenCompressed {
+		return fmt.Errorf("DbPutLockedBalanceEntryMappingsWithTxn: HODLer PKID "+
+			"length %d != %d", len(lockedBalanceEntry.HODLerPKID), btcec.PubKeyBytesLenCompressed)
+	}
+	if len(lockedBalanceEntry.ProfilePKID) != btcec.PubKeyBytesLenCompressed {
+		return fmt.Errorf("DbPutLockedBalanceEntryMappingsWithTxn: Profile PKID "+
+			"length %d != %d", len(lockedBalanceEntry.ProfilePKID), btcec.PubKeyBytesLenCompressed)
+	}
+
+	if err := DBSetWithTxn(txn, snap, _dbKeyForLockedBalanceEntry(lockedBalanceEntry),
+		EncodeToBytes(blockHeight, &lockedBalanceEntry)); err != nil {
+		return errors.Wrapf(err, "DbPutLockedBalanceEntryMappingsWithTxn: "+
+			"Problem adding locked balance entry to db")
+	}
+	return nil
+}
+
+func DbDeleteLockedBalanceEntryWithTxn(txn *badger.Txn, snap *Snapshot, lockedBalanceEntry LockedBalanceEntry) error {
+	// First check that a mapping exists. If one doesn't then there's nothing to do.
+	_, err := DBGetWithTxn(txn, snap, _dbKeyForLockedBalanceEntry(lockedBalanceEntry))
+	if err != nil {
+		return nil
+	}
+
+	// When a locked balance entry exists, delete the locked balance entry mapping.
+	if err := DBDeleteWithTxn(txn, snap, _dbKeyForLockedBalanceEntry(lockedBalanceEntry)); err != nil {
+		return errors.Wrapf(err, "DbDeleteRepostMappingsWithTxn: Deleting "+
+			"locked balance entry for HODLer PKID %s, Profile PKID %s, expiration timestamp %d",
+			lockedBalanceEntry.HODLerPKID.ToString(), lockedBalanceEntry.ProfilePKID.ToString(),
+			lockedBalanceEntry.ExpirationTimestampNanoSecs)
+	}
+	return nil
+}
+
+// LockedBalanceEntry Get Operations (Badger Reads)
+
+func DBGetLockedBalanceEntryForHODLerPKIDProfilePKIDExpirationTimestampNanoSecs(handle *badger.DB, snap *Snapshot,
+	hodlerPKID *PKID, profilePKID *PKID, expirationTimestamp int64) *LockedBalanceEntry {
+
+	var ret *LockedBalanceEntry
+	handle.View(func(txn *badger.Txn) error {
+		ret = DBGetLockedBalanceEntryForHODLerPKIDProfilePKIDExpirationTimestampNanoSecsWithTxn(
+			txn, snap, hodlerPKID, profilePKID, expirationTimestamp)
+		return nil
+	})
+	return ret
+}
+
+func DBGetLockedBalanceEntryForHODLerPKIDProfilePKIDExpirationTimestampNanoSecsWithTxn(txn *badger.Txn, snap *Snapshot,
+	hodlerPKID *PKID, profilePKID *PKID, expirationTimestamp int64) *LockedBalanceEntry {
+
+	key := _dbKeyForLockedBalanceEntry(LockedBalanceEntry{
+		HODLerPKID:                  hodlerPKID,
+		ProfilePKID:                 profilePKID,
+		ExpirationTimestampNanoSecs: expirationTimestamp,
+	})
+	lockedBalanceEntryBytes, err := DBGetWithTxn(txn, snap, key)
+	if err != nil {
```

Comment:
I found it a bit odd that we return something here when it's missing. Two comments here:
1. I'm not really sure what the point of this is if we're going to return the thing they passed in anyway basically.
2. If you definitely want this functionality, then normally what we do is check that err is "Not Found" or whatever the "Not Found" enum is. Otherwise, you run the risk of swallowing more serious errors. Maybe you can return nil if it's an error you don't recognize, and check it on the caller?

Reply:
Revisiting this at the end after completing my review. I still think it makes everything easier to read and less error-prone to not return a default value here.